components/experimental/src/dimension/units/formatter.rs

```diff
@@ -111,7 +119,15 @@ impl UnitsFormatter {
 
         let plural_rules = PluralRules::try_new_cardinal_unstable(provider, locale)?;
 
-        let unit_attribute = DataMarkerAttributes::try_from_str(unit)
+        let length = match options.width {
+            Width::Short => "short",
+            Width::Narrow => "narrow",
+            Width::Long => "long",
+        };
+
+        let attribute = length.to_string() + "-" + unit;
```

Comment:
issue: ideally we don't have to allocate here

Reply:
@sffc @robertbastian is there a good way to avoid the allocation here? I'm not sure if preallocating a stack buffer is that great either.

Reply:
I used buffer to solve this issue, which I think it is not ideal.

we can use constants to all the keys:
```
const LONG_METER: &'static DataMarkerAttributes =
        DataMarkerAttributes::from_str_or_panic("long-meter");
```

However, the number of the keys is so big.

Do you have ideas how to solve this issue ?

Reply:
It's necessary to build a string if there are multiple components of a data key attributes, _but_:

1. You should ideally use a stack buffer (for fixed length) or something like `smallstr` (for variable length)
2. ***BUT***, in this case, this is irrelevant because you should create separate data markers for the lengths.

Reply:
Yes, I know that we are planning to have them in different marker. But in this PR, I want to split them by length by using Auxiliary keys.

In the next PRs, we will have different markers for each length and category.

Reply:
This should happen before this PR, because then you won't have to submit allocating code. We this was already discussed in https://github.com/unicode-org/icu4x/pull/5134

Reply:
Add `TODO` and just use string. or `SmallVec` or `SmallStr`

Reply:
done